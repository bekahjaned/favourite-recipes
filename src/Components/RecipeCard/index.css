ul,
ol,
h4 {
  margin: 5px;
  font-family: "Open Sans", sans-serif;
}

li {
  margin-bottom: 2px;
}

h3 {
  margin: 5px;
  font-family: "Libre Baskerville", serif;
  text-decoration: underline;
}
