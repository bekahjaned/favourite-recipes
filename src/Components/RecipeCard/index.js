import React from "react";

import { RecipeWrapper } from "../../Elements/RecipeWrapper/";

import "./index.css";

class RecipeCard extends React.Component {
  render() {
    const ingredientList = this.props.ingredients.map((ingredient) => {
      return <li>{ingredient}</li>;
    });

    const methodList = this.props.method.map((step) => {
      return <li>{step}</li>;
    });

    return (
      <RecipeWrapper>
        <h3>{this.props.name}</h3>
        <div>
          <h4>Ingredients:</h4>
          <ul>{ingredientList}</ul>
        </div>
        <div>
          <h4>Method:</h4>
          <ol>{methodList}</ol>
        </div>
      </RecipeWrapper>
    );
  }
}

export default RecipeCard;
