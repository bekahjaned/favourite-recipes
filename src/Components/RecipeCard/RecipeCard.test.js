import React from "react";
import renderer from "react-test-renderer";

import RecipeCard from "./index.js";

it("renders Recipe Card", () => {
  // given
  const name = "name";
  const ingredients = ["one", "two", "three"];
  const method = ["one", "two", "three"];

  // when
  const tree = renderer
    .create(
      <RecipeCard name={name} ingredients={ingredients} method={method} />
    )
    .toJSON();

  // then
  expect(tree).toMatchSnapshot();
  expect(
    tree.children
      .find((child) => child.type === "div")
      .children.find((child) => child.type === "ul").children.length
  ).toEqual(ingredients.length);
});
