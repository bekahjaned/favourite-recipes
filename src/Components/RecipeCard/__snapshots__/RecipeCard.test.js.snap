// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Recipe Card 1`] = `
<div
  className="sc-AxjAm hDhnSf"
>
  <h3>
    name
  </h3>
  <div>
    <h4>
      Ingredients:
    </h4>
    <ul>
      <li>
        one
      </li>
      <li>
        two
      </li>
      <li>
        three
      </li>
    </ul>
  </div>
  <div>
    <h4>
      Method:
    </h4>
    <ol>
      <li>
        one
      </li>
      <li>
        two
      </li>
      <li>
        three
      </li>
    </ol>
  </div>
</div>
`;
